### DISCUSSION TITLE
How to get the device id in a multi-gpu host

### DISCUSSION THREAD

**Discussion by jrabary** [jax-ml/jax#8073] [Q&A] [answered]
Dear all,

I need to get the id of the device where my computation takes place.

Say for example, I have 8 gpu on my host and I need to multiply a tensor by a specific number per GPU 
like `a * (GPU_ID + 1)`.

There is a `jax.device_count` but  I'd rather need a `jax.device_id` or `jax.local_device_id`

**Comment by jrabary** [comment]
More precisely, I'm looking for the jax implementation of the following tensorflow function

```
replica_context = tf.distribute.get_replica_context()
id = replica_context.replica_id_in_sync_group
```

**Comment by skye** [comment, marked as answer]
I assume you're using pmap? One option is to pass `jnp.arange(jax.device_count())` into the pmap, so every GPU will get a different value. Alternatively, you can use [jax.lax.axis_index](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.axis_index.html#jax.lax.axis_index).

**Reply by jrabary** [reply]
Thank you, this is what I need.
